File: uwp/sited/__CacheBlock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace org.noear.sited {
   internal class __CacheBlock {
        public String value = null;
        public DateTime time = DateTime.MinValue;
        
        public bool isOuttime(SdNode config) {
            if (time == null || value == null) {
                return true;
            }
            else {
                if (config.cache == 1)
                    return false;
                else {
                    long seconds = ( DateTime.Now - time ).Seconds;
                    return seconds > config.cache;
                }
            }
        }
    }
}

File: android/sited/src/main/java/org/noear/sited/__CacheBlock.java
package org.noear.sited;

/**
 * Created by yuety on 15/10/10.
 */

import java.util.Date;

class __CacheBlock {
    public String value;
    public Date time;

    public boolean isOuttime(SdNode config) {
        if(time==null || value == null){
            return true;
        }else {
            if(config.cache == 1)
                return false;
            else {
                long seconds = (new Date().getTime() - time.getTime()) / 1000;
                return seconds > config.cache;
            }
        }
    }
}

